dm/syncer/syncer.go

```diff
@@ -1483,15 +1475,36 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 	if err != nil {
 		return err
 	}
-	if s.cfg.Mode == config.ModeAll && fresh {
-		delLoadTask = true
-		flushCheckpoint = true
-		err = s.loadTableStructureFromDump(ctx)
+	if fresh {
+		if s.cfg.Mode == config.ModeAll {
+			delLoadTask = true
+			flushCheckpoint = true
+			err = s.loadTableStructureFromDump(ctx)
+			if err != nil {
+				tctx.L().Warn("error happened when load table structure from dump files", zap.Error(err))
+				cleanDumpFile = false
+			}
+		}
+		tbls := s.optimist.Tables()
```

Comment:
why always fetch tbls regardless of whether user open the optimist?

Reply:
update